six/modules/c++/scene/source/ProjectionModel.cpp

```diff
@@ -62,11 +62,19 @@ ProjectionModel(const Vector3& slantPlaneNormal,
     mSlantPlaneNormal(slantPlaneNormal),
     mSCP(scp),
     mARPPoly(arpPoly),
-    mARPVelPoly(mARPPoly.derivative()),
     mTimeCOAPoly(timeCOAPoly),
     mLookDir(lookDir),
     mErrors(errors)
 {
+    try
```

Comment:
This will throw when?  When the poly is empty so order() throws?  If we really need to guard against that - I take it you had some confusing exception message you tracked down to this - can you make it one standalone function in the unnamed namespace that does this?  This way you can reuse the code and still keep things in the initializer list.  The function would take in a string for the poly name.